Formula/w/whisper-cpp.rb

```diff
@@ -0,0 +1,19 @@
+class WhisperCpp < Formula
+  desc "Port of OpenAI's Whisper model in C/C++"
+  homepage "https://github.com/ggerganov/whisper.cpp"
+  url "https://github.com/ggerganov/whisper.cpp/archive/refs/tags/v1.4.0.tar.gz"
+  sha256 "b2e34e65777033584fa6769a366cdb0228bc5c7da81e58a5e8dc0ce94d0fb54e"
+  license "MIT"
+  head "https://github.com/ggerganov/whisper.cpp.git", branch: "master"
+
+  depends_on "cmake" => :build
+  def install
+    system "make"
+    bin.install "main" => "whisper-cpp"
+  end
+
+  test do
+    system "#{bin}/whisper-cpp", "--help"
+    assert_equal 0, $CHILD_STATUS.exitstatus, "whisper-cpp failed with exit code #{$CHILD_STATUS.exitstatus}"
```

Comment:
need a more substantial test

Reply:
The only real way to test it would be to transcribe the sample file or do `./main bench`, but both require downloading a model...

`ggml-model-whisper-tiny.en-q5_1.bin` is a reduced precision smallest English-only model the smallest, which is about 31M, so I guess it might suffice for a test?

Reply:
I've implemented a test with a dummy model. The logs said no tensors were loaded, but my laptop did make some coil whine so it must have done some calculations... I might need to intentionally produce a miscompiled binary to test it for sure...

I hope it's ok that it's redownloading sample files from the repo, as I saw no way a test could access a build environment's data.

Reply:
The dummy models perform the entire computation as the original models - it just uses random data instead of the real model weights.